include/constants/battle_ai.h

```diff
@@ -59,6 +59,8 @@
 #define AI_FLAG_SMART_SWITCHING       (1 << 15)  // AI includes a lot more switching checks
 #define AI_FLAG_ACE_POKEMON           (1 << 16)  // AI has an Ace Pokemon. The last Pokemon in the party will not be used until it's the last one remaining.
 #define AI_FLAG_OMNISCIENT            (1 << 17)  // AI has full knowledge of player moves, abilities, hold items
+#define AI_FLAG_SMART_MON_CHOICES     (1 << 18)  // AI will make smarter decisions when choosing which mon to send out mid-battle. Pairs very well with AI_FLAG_SMART_SWITCHING.
```

Comment:
Because unlike smart switching this flag adds significant delay before the player's turn (~0.4s in a single battle against 6 mons vs vanilla's 0.2s), less delay with fewer mons.

I wanted people to be able to opt in to smart switching (AI makes better decisions about _when_ to switch) separately from smart mon choices (AI makes better decisions about _what to switch into_) so people could more modularly scale trainer difficulty and also opt in to the delay separately from the switch decision improvements if they wanted to.

Reply:
Okay, that's fair. Btw, how do you measure the delay? I have some ideas to test which could potentially reduce it further 👀

Reply:
It's not especially analytical, I record the same battle with the changes and without the change with OBS, bring both into Windows Video editor, and chop the clip to find the delay such that the last frame of "Go! Pokemon" and the first frame where the menu pops are are the boundaries.

I haven't done deeply rigorous testing in about a week, but somewhere around ~0.43s is very consistent for a full team of 6 mons, in a single battle, with smart switching and mon smart mon choices flags :)

Reply:
This is good, but it takes too much time. If I get some free time, I'll try to come up with a better in-game solutions, so it prints a debug message or sth to measure the delay.

Reply:
That'd be super convenient, let me know if you get to it! Happy to continue measuring with Windows Video Editor in the meantime lol